src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js

```diff
@@ -283,23 +283,31 @@ export default class StandardRouteWaypointModel extends BaseModel {
      * @for StandardRouteWaypointModel
      * @method clonePositionFromFix
      * @param fixCollection {FixCollection}
+     * @return {StandardRouteWaypointModel}
      * @private
+     * @chainable
      */
     clonePositionFromFix() {
-        if (this.name[0] === VECTOR_WAYPOINT_PREFIX) {
+        const isFlyOverWaypoint = this.name.indexOf(FLY_OVER_WAYPOINT_PREFIX) !== -1;
+        const isHoldWaypoint = this.name.indexOf(HOLD_WAYPOINT_PREFIX) !== -1;
+        const isVectorWaypoint = this.name.indexOf(VECTOR_WAYPOINT_PREFIX) !== -1;
```

Comment:
This looks good!  As I'm reading this, though, don't we already have helpers for these in the (poorly named) `RouteModel` as static methods?